### PR TITLE
Build 10.34 and use CMake and build for Windows

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
-1 to moving to cmake. This would break existing software because the autoconf system has SONAME embedded and the cmake build doesn't.

**Comment by xhochy** [comment]
> -1 to moving to cmake. This would break existing software because the autoconf system has SONAME embedded and the cmake build doesn't.

I'm working on a patch for CMake as the autoconf system also ships support for `pkg-config` while the CMake version doesn't. I would prefer to use CMake as this would make the Windows build simpler.

**Comment by isuruf** [comment]
You should work on setting the SOVERSION, VERSION, MACHO_COMPATIBILITY_VERSION, MACHO_CURRENT_VERSION properties on the cmake libraries. Otherwise -1 to change.

**Comment by isuruf** [comment]
I've created a list of TODOs here https://github.com/conda-forge/conda-forge.github.io/pull/1036/files when switching

**Comment by xhochy** [comment]
@conda-forge/pcre2 @isuruf This is ready for review.

File: recipe/cmake-configure-alignment.patch [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+-ADD_LIBRARY(pcre2-8 ${PCRE2_HEADERS} ${PCRE2_SOURCES} ${PROJECT_BINARY_DIR}/config.h)
+-SET_PROPERTY(TARGET pcre2-8
+-  PROPERTY COMPILE_DEFINITIONS PCRE2_CODE_UNIT_WIDTH=8)
++  ADD_LIBRARY(pcre2-8 ${PCRE2_HEADERS} ${PCRE2_SOURCES} ${PROJECT_BINARY_DIR}/config.h)
++
++SET_TARGET_PROPERTIES(pcre2-8 PROPERTIES
++  COMPILE_DEFINITIONS PCRE2_CODE_UNIT_WIDTH=8
++  MACHO_COMPATIBILITY_VERSION 10.0.0
++  MACHO_CURRENT_VERSION 10.0.0
++  VERSION 0.9.0
```
Comment (isuruf) [inline recipe/cmake-configure-alignment.patch:24]: Shouldn't this be `0.8.0`?

Comment (xhochy) [reply recipe/cmake-configure-alignment.patch:24]: No, they increased it to 9. See also the Debian list of files: https://packages.debian.org/de/sid/amd64/libpcre2-8-0/filelist Debian builds with autotools: https://salsa.debian.org/debian/pcre2/-/blob/master/debian/rules#L26 I'm a bit confused that ABI Laboratory reports differently.

Comment (isuruf) [reply recipe/cmake-configure-alignment.patch:24]: Okay. I see now I was looking at the wrong source.

File: recipe/cmake-configure-alignment.patch [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
++  SOVERSION 0)
```
Comment (isuruf) [inline recipe/cmake-configure-alignment.patch:25]: Shouldn't all these versions be parsed from `configure.ac` like `PCRE2_MAJOR` is parsed just a few lines above in the CMakeLists.txt

Comment (xhochy) [reply recipe/cmake-configure-alignment.patch]: Refreshed my autotools version computation knowledge and all versions are now parsed from `configure.ac`.

File: recipe/cmake-configure-alignment.patch [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
++  VERSION ${LIBPCRE2_8_VERSION}
++  SOVERSION ${LIBPCRE2_8_SOVERSION})
+ SET(targets ${targets} pcre2-8)
+ ADD_LIBRARY(pcre2-posix ${PCRE2POSIX_HEADERS} ${PCRE2POSIX_SOURCES})
+-SET_PROPERTY(TARGET pcre2-posix
+-  PROPERTY COMPILE_DEFINITIONS PCRE2_CODE_UNIT_WIDTH=8)
++SET_TARGET_PROPERTIES(pcre2-posix PROPERTIES
++  COMPILE_DEFINITIONS PCRE2_CODE_UNIT_WIDTH=8
++  MACHO_COMPATIBILITY_VERSION ${LIBPCRE2_POSIX_MACHO_VERSION}
++  MACHO_CURRENT_VERSION ${LIBPCRE2_POSIX_MACHO_VERSION}
```
Comment (isuruf) [inline recipe/cmake-configure-alignment.patch:65]: AFAIR, `compatibility_version = current + 1` and `current_version = (current + 1).revision`

Comment (xhochy) [reply recipe/cmake-configure-alignment.patch]: Checked on master of `pcre2` that this formula holds, only references I found only are other PRs fixing this. Not even the autotools mythbuster seems to mention this.

**Comment by xhochy** [comment]
Contributed both patches upstream (no review yet) so we can get hopefully rid of them in the future:

- https://bugs.exim.org/show_bug.cgi?id=2557
- https://bugs.exim.org/show_bug.cgi?id=2558

**Comment by xhochy** [comment]
@conda-forge/pcre2 @isuruf Any more comments?

**Comment by isuruf** [comment]
+1 to merge if @conda-forge/pcre2 is fine with it.

**Comment by mariusvniekerk** [comment]
Think i'm actually on that group as the last person to touch it during the great compiler migration.  Looks good to me.  If @isuruf is happy i'm happy

**Comment by bgruening** [comment]
I pinged @dfornika. :+1: from my side.

**Comment by dfornika** [comment]
👍 for me. Should I press 'Squash and merge'? It's been a while since I merged something on conda-forge. Is there a bot or something that handles merges like there is on bioconda?

**Comment by dfornika** [comment]
Thanks @xhochy @mariusvniekerk @isuruf @bgruening  for your work on this.

**Pull request merged**